src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +272,467 @@ define(function (require, exports, module) {
+
+            MainViewManager._moveView("first-pane", "second-pane", file, 0, true);
+
+            // Now refresh the project tree by asking
+            // it to rebuild the UI.
+            WorkingSetView.refresh(true);
+        }
+
+        CommandManager.execute(Commands.FILE_OPEN, { fullPath: prefsPath, paneId: "second-pane"});
+
+    }
+
+    function _isSupportedPrefType(prefType) {
+        if (prefType !== undefined &&
+                (prefType === "number" ||
+                prefType === "boolean" ||
+                prefType === "string"  ||
+                prefType === "array"   ||
+                prefType === "object")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // This method tries to deduce the preference type
+    // based on various parameters like objects initial
+    // value, object type, object's type property.
+    function _getObjType(prefObj) {
+
+        var prefType = "undefined";
+
+        if (prefObj) {
+            // check the type parameter.
+            var _type = prefObj.type;
+            if (typeof (_type) !== "undefined") {
+                prefType = prefObj.type.toLowerCase();
+                // make sure the initial property's
+                // object type matches to that of 'type' propety.
+                if (prefObj.initial !== undefined) {
+
+                    if (Array.isArray(prefObj.initial)) {
+                        _type = "array";
+                    } else {
+                        var _initialType = typeof (prefObj.initial);
+                        _initialType = _initialType.toLowerCase();
+                        if (_type !== _initialType) {
+                            _type = _initialType;
+                        }
+                    }
+                }
+            }
+
+            if (_type) {
+                // preference object's type
+                // is defined. Check if that is valid or not.
+                prefType = _type;
+                if ((prefType !== "number"  &&
+                     prefType !== "boolean" &&
+                     prefType !== "string"  &&
+                     prefType !== "array"   &&
+                     prefType !== "object")) {
+                    prefType = "undefined";
+                }
+            } else if (Array.isArray(prefObj)) {
+                // Check if the object itself
+                // is an array, in which case
+                // we log the default.
+                prefType = "array";
+            } else if (typeof (prefObj.initial) !== "undefined"  ||
+                       typeof (prefObj.keys) !== "undefined") {
+
+                // OK looks like this preference has
+                // no explicit type defined. instead
+                // it needs to be deduced from initial/keys
+                // variable.
+                var _prefVar;
+                if (typeof (prefObj.initial) !== "undefined") {
```

Comment:
You don't need `typeof` to check for `undefined`, a more simple `if (prefObj.initial !== undefined)` suffices.

This is not only true for this particular line, but for the overall changes.